cmake/doctest.cmake

```diff
@@ -0,0 +1,30 @@
+find_package(doctest 2.4.8 QUIET)
```

Comment:
Please be more careful if you insist to on using minimum supported versions.
Version 2.4.8 doesn't support the `doctest_with_main` cmake alias that you're using.

Reply:
Right, I also don't like it, but that's what Ubuntu-22.04 ships.
After much suffering, I am leaning towards dropping support for this version.